service/matching/backlog_manager_test.go

```diff
@@ -154,3 +157,164 @@ func TestTaskWriterShutdown(t *testing.T) {
 	err = tlm.backlogMgr.SpoolTask(&persistencespb.TaskInfo{})
 	require.Error(t, err)
 }
+
+func TestApproximateBacklogCountIncrement_taskWriterLoop(t *testing.T) {
+	controller := gomock.NewController(t)
+	backlogMgr := newBacklogMgr(controller, false)
+
+	// Add tasks on the taskWriters channel
+	backlogMgr.taskWriter.appendCh <- &writeTaskRequest{
+		taskInfo: &persistencespb.TaskInfo{
+			ExpiryTime: timestamp.TimeNowPtrUtcAddSeconds(3000),
+			CreateTime: timestamp.TimeNowPtrUtc(),
+		},
+		responseCh: make(chan<- *writeTaskResponse),
+	}
+
+	require.Equal(t, backlogMgr.db.getApproximateBacklogCount(), int64(0))
```

Comment:
order should be Equal(expected, actual)
```suggestion
	require.Equal(t, int64(0), backlogMgr.db.getApproximateBacklogCount())
```